tests/test_exacttarget_field_selection.py

```diff
@@ -0,0 +1,133 @@
+            'UnsubscribedDate',
+            'PrimarySMSAddress', # not retrievable
+            'PrimaryEmailAddress', # not retrievable
+            'PartnerProperties', # not retrievable
+            'SubscriberTypeDefinition', # not retrievable
+            'Addresses', # not retrievable
+            'ListIDs',
+            'Locale', # not retrievable
+            'PrimarySMSPublicationStatus', # not retrievable
+            'ModifiedDate'], # not retrievable
+        'list_send': [
+            'CreatedDate', # not retrievable
+            'CustomerKey', # not retrievable
+            'ID',
+            'PartnerProperties', # not retrievable
+            'ModifiedDate'], # not retrievable
+        'folder': [
+            'Type',
+            'PartnerProperties'],
+        'email': [
+            '__AdditionalEmailAttribute1', # not retrievable
+            '__AdditionalEmailAttribute3', # not retrievable
+            'SyncTextWithHTML', # not retrievable
+            'PartnerProperties', # not retrievable
+            '__AdditionalEmailAttribute5', # not retrievable
+            'ClonedFromID',
+            '__AdditionalEmailAttribute4', # not retrievable
+            '__AdditionalEmailAttribute2'], # not retrievable
+        'content_area': [
+            # most of them are included in the 'Content' data
+            'BackgroundColor', # not retrievable
+            'Cellpadding', # not retrievable
+            'HasFontSize', # not retrievable
+            'BorderColor', # not retrievable
+            'BorderWidth', # not retrievable
+            'Width', # not retrievable
+            'IsLocked', # not retrievable
+            'Cellspacing', # not retrievable
+            'FontFamily'] # not retrievable
+    }
+
+    # fields not to select
+    non_selected_fields = {
+            "campaign": ["createdDate", "campaignCode", "description"],
+            "content_area": ["Name"],
+            "data_extension.test emails": ["name", "email"],
+            "data_extension.This is a test": ["Birthday"],
+            "data_extension.my_test": ["Address"],
+            "data_extension.test 1": ["name"],
+            "email": ["Name", "CharacterSet", "HasDynamicSubjectLine", "EmailType"],
+            "event": ["EventType"],
+            "folder": ["Name", "ContentType", "Description", "ObjectID"],
+            "list": ["ListName", "Category", "Type"],
+            "list_send": ["MissingAddresses", "ExistingUndeliverables", "HardBounces", "NumberDelivered"],
+            "list_subscriber": ["Status", "ObjectID", "ListID"],
+            "send": ["Status", "EmailName", "FromAddress", "IsMultipart"],
+            "subscriber": ["Status", "EmailAddress", "SubscriberKey", "PartnerKey"]
+        }
+
+    def name(self):
+        return "tap_tester_exacttarget_field_selection"
+
+    def test_run(self):
+        # run test with all fields of stream 'selected = False'
+        self.run_test(only_automatic_fields=True)
```

Comment:
If you are going to test automatic fields in this test, please include all the standard assertions for that test https://github.com/stitchdata/tap-tester/blob/master/reference/expectations/saas_taps.md#test_automatic_fieldspy. Looks like you are only missing the check for unique records.

Reply:
Added verification of unique records.